Source/Core/GoogleEarthEnterpriseMetadata.js

```diff
@@ -496,8 +502,23 @@ define([
             }
         });
 
-        return resource.fetchArrayBuffer()
-            .then(function(buf) {
+        if (!defined(dbrootParserPromise)) {
+            var url = buildModuleUrl('ThirdParty/google-earth-dbroot-parser.js');
+            var oldValue = window.cesiumGoogleEarthDbRootParser;
+            dbrootParserPromise = loadAndExecuteScript(url)
+                .then(function() {
+                    dbrootParser = window.cesiumGoogleEarthDbRootParser(protobufMinimal);
+                    if (defined(oldValue)) {
+                        window.cesiumGoogleEarthDbRootParser = oldValue;
```

Comment:
It looks like we're using `window.cesiumGoogleEarthDbRootParser` for both the loaded script, and the function evaluated with `protobufMinimal`. Which means you have to set back the `oldValue` after evaluating. Would this be clearer if we just loaded the module to  `window.cesiumGoogleEarthDbRootModule` or something similar?

Reply:
I'm not sure what you mean.  Here's what's currently happening

1. We get the old value of `cesiumGoogleEarthDbRootParser` on the off-chance it exists
2. We load the script, which defines `cesiumGoogleEarthDbRootParser`.
3. We call the `cesiumGoogleEarthDbRootParser` function and store the result into `dbrootParser` (module scoped)
4. We set back the old value of `cesiumGoogleEarthDbRootParser` or delete it completely if it didn't exist to begin with.

Reply:
Got it, sorry for the confusion. Makes sense.